src/main/java/org/sonarlint/intellij/ui/vulnerabilities/TaintVulnerabilitiesPanel.kt

```diff
@@ -307,6 +313,39 @@ class TaintVulnerabilitiesPanel(private val project: Project) : SimpleToolWindow
         currentStatus?.let { populateTrees(it) }
     }
 
+    fun doesTaintExist(taintKey: String): LocalTaintVulnerability? {
+        val issue = treeBuilder.findTaintByKey(taintKey)?.let {
+            oldTreeBuilder.findTaintByKey(taintKey)
+        }
+        return issue
+    }
+
+    fun <T: Finding> trySelectFilteredTaintVulnerability(taintVulnerability: LocalTaintVulnerability?, showFinding: ShowFinding<T>) {
```

Comment:
The name of the method made me think, but this is more a general question: what if the issue/hotspot/taint exists and was detected but is filtered out in the views? For example if the finding is resolved. I saw that you already check in both trees in the case of new code period so this should work, I'm not sure about resolved findings

Reply:
The `findTaintByKey` method looks at any issue in the tree, regardless of the filters. So this is effectively working, but if it was resolved and the filter is not enabled, the issue won't appear in the list (but still, it will be highlighted and the rule description is shown, going with the flow of a regular finding, not a server finding).

This is an improvement that I'd like to have: when opening the finding in the issue, we could have the finding selected/focused in the list, so the user is aware of which one it is. But it also requires a bit more work because if the finding is filtered out, we need the user to enable the filters. We did something similar for security hotspot, we showed a notification with an action to clear the filtering. But for hotspot the workflow is different...